<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PromptlyDialogBase</class>
 <widget class="QDialog" name="PromptlyDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>700</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Promptly</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QGroupBox {
    border: 1px solid #b8b8b8;
    border-radius: 6px;
    margin-top: 12px;
    font-weight: bold;
    background-color: #fafafa;
}
QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 5px;
}
QPushButton {
    background-color: #2d8bca;
    color: white;
    border-radius: 4px;
    padding: 6px 12px;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #3498db;
}
QPushButton:pressed {
    background-color: #2980b9;
}
QPushButton:disabled {
    background-color: #cccccc;
    color: #666666;
}
QTabWidget::pane {
    border: 1px solid #b8b8b8;
    border-radius: 4px;
}
QTabBar::tab {
    background-color: #e6e6e6;
    border: 1px solid #b8b8b8;
    border-bottom-color: transparent;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    min-width: 8ex;
    padding: 5px 10px;
}
QTabBar::tab:selected {
    background-color: #ffffff;
    border-bottom-color: transparent;
}
QLabel#labelStatus {
    font-style: italic;
    color: #666666;
}
QDialog {
    background-color: #f5f5f5;
}
QComboBox, QLineEdit {
    padding: 5px;
    border: 1px solid #b8b8b8;
    border-radius: 4px;
}
QPlainTextEdit {
    border: 1px solid #b8b8b8;
    border-radius: 4px;
    background-color: #ffffff;
}
</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1,0">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>12</number>
   </property>
   <property name="topMargin">
    <number>12</number>
   </property>
   <property name="rightMargin">
    <number>12</number>
   </property>
   <property name="bottomMargin">
    <number>12</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutHeader">
     <item>
      <widget class="QLabel" name="labelLogo">
       <property name="maximumSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap>icons/icon.png</pixmap>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelTitle">
       <property name="font">
        <font>
         <pointsize>14</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Promptly</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="handleWidth">
      <number>8</number>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="topWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>4</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QTabWidget" name="settingsTabs">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tabSettings">
          <attribute name="title">
           <string>LLM Settings</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="labelProvider">
               <property name="text">
                <string>LLM Provider:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1" colspan="3">
              <widget class="QComboBox" name="comboBoxProvider">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <item>
                <property name="text">
                 <string>Ollama</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>OpenAI</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>OpenRouter</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Anthropic</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Custom</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="labelModel">
               <property name="text">
                <string>Model:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1" colspan="3">
              <widget class="QLineEdit" name="lineEditModel">
               <property name="text">
                <string>qwen2.5-coder:32b-instruct-q5_K_M</string>
               </property>
               <property name="placeholderText">
                <string>Model name or ID</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="labelApiKey">
               <property name="text">
                <string>API Key:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1" colspan="3">
              <widget class="QLineEdit" name="lineEditApiKey">
               <property name="echoMode">
                <enum>QLineEdit::Password</enum>
               </property>
               <property name="placeholderText">
                <string>Not required for Ollama, required for other providers</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="labelServer">
               <property name="text">
                <string>API Endpoint:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1" colspan="3">
              <widget class="QLineEdit" name="lineEditServer">
               <property name="text">
                <string>http://localhost:11434/api/generate</string>
               </property>
               <property name="placeholderText">
                <string>API endpoint URL</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="labelTemperature">
               <property name="text">
                <string>Temperature:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QDoubleSpinBox" name="doubleSpinBoxTemperature">
               <property name="decimals">
                <number>1</number>
               </property>
               <property name="maximum">
                <double>2.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
               <property name="value">
                <double>0.700000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="2">
              <widget class="QLabel" name="labelMaxTokens">
               <property name="text">
                <string>Max Tokens:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="3">
              <widget class="QSpinBox" name="spinBoxMaxTokens">
               <property name="minimum">
                <number>100</number>
               </property>
               <property name="maximum">
                <number>100000</number>
               </property>
               <property name="singleStep">
                <number>100</number>
               </property>
               <property name="value">
                <number>4000</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabResources">
          <attribute name="title">
           <string>Resources</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayoutResources">
             <item>
              <widget class="QGroupBox" name="groupBoxDatabase">
               <property name="title">
                <string>Database Reference</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_11">
                <item>
                 <widget class="QLabel" name="labelDatabaseDesc">
                  <property name="text">
                   <string>Select a database to use as reference for SQL queries</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout">
                  <item>
                   <widget class="QComboBox" name="comboBoxSqlDatabase">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <item>
                     <property name="text">
                      <string>None</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="pushButtonRefreshDatabases">
                    <property name="maximumSize">
                     <size>
                      <width>30</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>↻</string>
                    </property>
                    <property name="toolTip">
                     <string>Refresh Database List</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBoxLayer">
               <property name="title">
                <string>Layer Reference</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_12">
                <item>
                 <widget class="QLabel" name="labelLayerDesc">
                  <property name="text">
                   <string>Select a layer to use for metadata reference</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayoutLayer">
                  <item>
                   <widget class="QComboBox" name="comboBoxLayer">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <item>
                     <property name="text">
                      <string>None</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="pushButtonRefreshLayers">
                    <property name="maximumSize">
                     <size>
                      <width>30</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>↻</string>
                    </property>
                    <property name="toolTip">
                     <string>Refresh Layer List</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxPrompt">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Prompt</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPlainTextEdit" name="plainTextEditPrompt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>120</height>
             </size>
            </property>
            <property name="placeholderText">
             <string>Enter your QGIS-related prompt here...</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayoutSend">
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButtonSend">
              <property name="minimumSize">
               <size>
                <width>180</width>
                <height>40</height>
               </size>
              </property>
              <property name="text">
               <string>Send to LLM</string>
              </property>
              <property name="icon">
               <iconset>
                <normaloff>icons/send.png</normaloff>icons/send.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButtonCancel">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>40</height>
               </size>
              </property>
              <property name="text">
               <string>Cancel</string>
              </property>
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="toolTip">
               <string>Cancel the current LLM request</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="bottomWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>6</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QGroupBox" name="groupBoxResponse">
         <property name="title">
          <string>LLM Response</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QTabWidget" name="tabWidgetResponse">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>200</height>
             </size>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tabFullResponse">
             <attribute name="title">
              <string>Full Response</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QPlainTextEdit" name="plainTextEditFullResponse">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
                <property name="placeholderText">
                 <string>LLM response will appear here...</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabCode">
             <attribute name="title">
              <string>Executable Code</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QPlainTextEdit" name="plainTextEditCode">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="placeholderText">
                 <string>Extracted code will appear here. You can edit it before execution...</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutExecute">
                <item>
                 <spacer name="horizontalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="pushButtonExecute">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>40</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Execute Code</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/play.png</normaloff>icons/play.png</iconset>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabErrorLog">
             <attribute name="title">
              <string>Error Log</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QPlainTextEdit" name="plainTextEditErrorLog">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
                <property name="placeholderText">
                 <string>Execution errors will appear here...</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutFix">
                <item>
                 <spacer name="horizontalSpacer_4">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="pushButtonFixCode">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>40</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Fix Code with LLM</string>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>icons/repair.png</normaloff>icons/repair.png</iconset>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutFooter">
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="value">
        <number>0</number>
       </property>
       <property name="textVisible">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelStatus">
       <property name="text">
        <string>Status: Ready</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="button_box">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Close</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>button_box</sender>
   <signal>accepted()</signal>
   <receiver>PromptlyDialogBase</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_box</sender>
   <signal>rejected()</signal>
   <receiver>PromptlyDialogBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui> 